File: ContactListAndroid/app/src/main/java/com/example/app/TestExecuter.java
package com.example.app;

import java.util.ArrayList;

/**
 * Created by Egorbo on 05.03.14.
 */
public class TestExecuter {

    private long startTime = 0;
    private final int TestsCount = 5;

    public String Run(Test test){
        ArrayList<Long> durationsList = new ArrayList<Long>(TestsCount);
        for (int i = 0; i < TestsCount; i++){
            StartTimer();
            test.Run();
            durationsList.add(GetElapsedTime());
        }
        long first = durationsList.get(0);
        long fastest = first;
        long slowest = first;
        long sum = 0;

        for (int i = 0; i < durationsList.size(); i++){
            long current = durationsList.get(i);
            if (current < fastest){
                fastest = current;
            }
            if (current > slowest){
                slowest = current;
            }
            sum += current;
        }

        double average = (double)(sum - fastest - slowest) / (durationsList.size() - 2.0);

        return String.format("First: %d, Fastest: %d, Slowest: %d, Average: %d", first, fastest, slowest, (long)average);
    }

    private void StartTimer() {
        startTime = System.currentTimeMillis();
    }

    private long GetElapsedTime() {
        long stopTime = System.currentTimeMillis();
        long elapsedTime = stopTime - startTime;
        return elapsedTime;
    }
}

File: ContactListXamarin/ContactListXamarin/TestExecuter.cs
using System.Collections.Generic;
using System.Diagnostics;
using System.Linq;

namespace ContactListXamarin
{
    public class TestExecuter
    {
        public const int TestsCount = 5;
        private readonly Stopwatch _stopwatch = new Stopwatch();

        public string Run(ITest test)
        {
            var timeSpans = new List<long>(TestsCount);
            for (int i = 0; i < TestsCount; i++)
            {
                _stopwatch.Restart();
                test.Run();
                _stopwatch.Stop();
                timeSpans.Add(_stopwatch.ElapsedMilliseconds);
            }
            long first = timeSpans.First();
            long max = timeSpans.Max();
            long min = timeSpans.Min();
			long avg = (timeSpans.Sum() - max - min)/(timeSpans.Count - 2);
            return string.Format("First: {0}, Fastest: {1}, Slowest: {2}, Average: {3}", first, min, max, avg);
        }
    }
}